state/factory/workingsetstore.go

```diff
@@ -33,24 +36,49 @@ func (store *workingSetStoreCommon) Filter(ns string, cond db.Condition, start,
 }
 
 func (store *workingSetStoreCommon) WriteBatch(bat batch.KVStoreBatch) error {
-	return store.flusher.KVStoreWithBuffer().WriteBatch(bat)
+	if err := store.flusher.KVStoreWithBuffer().WriteBatch(bat); err != nil {
+		return errors.Wrap(err, "failed to write batch")
+	}
+	if !store.committed {
+		return nil
+	}
+	return store.flusher.Flush()
 }
 
 func (store *workingSetStoreCommon) Put(ns string, key []byte, value []byte) error {
-	return store.flusher.KVStoreWithBuffer().Put(ns, key, value)
+	if err := store.flusher.KVStoreWithBuffer().Put(ns, key, value); err != nil {
+		return errors.Wrap(err, "failed to put value")
+	}
+	if !store.committed {
+		return nil
+	}
+	return store.flusher.Flush()
```

Comment:
when it gets here, the workingsetStore is already committed and `flusher.Flush()` called at L77. so in what use-case `Put()` will be called after committed, and why in such case need to call `Flush()` again?

Reply:
also `func (store *factoryWorkingSetStore) Put()` call this and do extra tlt handling
```
func (store *factoryWorkingSetStore) Put(ns string, key []byte, value []byte) error {
	store.workingSetStoreCommon.Put(ns, key, value)
	nsHash := hash.Hash160b([]byte(ns))

	return store.tlt.Upsert(nsHash[:], toLegacyKey(key), value)
}
```
does the tlt require special handling?

and same for `Delete()`

Reply:
as we discussed, after being committed, this `workingSetStoreCommon` functionally becomes a `KVStore` (for the next block N+1's workingset), so Put should write to underlying DB, that's correct
On the other hand, it is also a workingset's store, after being committed, it should not write to DB anymore, that's why the behavior here is somewhat confusing.
A better approach is to wrap it into another struct, similar to the recently `readOnlyKV` check-in, small change and clearly shows the intention, maybe a later PR to improve